core/src/test/java/org/bitcoinj/core/TransactionTest.java

```diff
@@ -589,8 +592,7 @@ public void testHashForSignatureThreadSafety() throws Exception {
                 new byte[0],
                 Transaction.SigHash.ALL.byteValue())
                 .toString();
-
-        final Callable<Void> c = () -> {
+        final Supplier<Void> supplier = () -> {
```

Comment:
I wonder if we should go back to a Runnable at this point? Would have the advantage of not having to return a dummy `null` value.

Reply:
Well it's only a test, so it's not that important how we write it.

But you had specifically asked me what was the best way to do it with JDK 8, so I tried my best to do it with CompletableFuture.

Personally, I would go with CompletableFuture despite the warts. We could move the genericArray method to a utility class in "core" and/or add an allOf(List) method to one of our CompletableFuture utility classes.

Reply:
And the Void/null thing is "idiomatic" for CompletableFuture -- I've seen it before and I'm used to it now. (Though it did bite me in a generic whenComplete lambda that I wrote)

Reply:
Oops, we can use `Runnable` with `CompletableFuture`

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#runAsync-java.lang.Runnable-

Reply:
Yes, that's what I was thinking of. Recently I saw [this table](https://stackoverflow.com/a/65772832) and thought maybe those 4 interface types are now the types to go. Do you think otherwise?

Reply:
The table looks good! 

Supplier is preferable to Callable because it doesn't throw checked exceptions.

I didn't realize Runnable was so officially lambda-friendly.